### DISCUSSION TITLE
I've compiled a version for 5.3.0 preview here

### DISCUSSION THREAD

**Discussion by courcreate** [microdee/OpenUnrealSpaceMouse#93] [Show and tell]
[Spacemouse5-3-0.zip](https://github.com/microdee/UE4-SpaceMouse/files/12399091/Spacemouse5-3-0.zip)

**Comment by microdee** [comment]
release is coming, I need to fix a simple compile error for 5.3

**Comment by onyxlee** [comment]
Thanks guys. I just tested both of your releases on UE 5.3 official, both of them don't work. Look forward to test a new version.

**Comment by onyxlee** [comment]
No rush, guys. Any updates? Just wanna thank you for your contribution to the community!💯

**Comment by microdee** [comment]
According to my tests this release should work https://github.com/microdee/UE4-SpaceMouse/releases/download/1.2.9/SpaceMouse-Win64-1.2.9.5.3-PreBuilt.zip

If it still doesn't work, can you elaborate more on the symptoms?

**Reply by onyxlee** [reply]
Thank you so much for taking care of us, Microdee! I just tested it in UE 5.3.1, It gave me this error message:

The following modules are missing or built with a different engine version:
HIDUE
SpaceMouseReader
SpaceMouseRuntime
SpaceMouseEditor

Engine modules cannot be compiled at runtime. Please build through your IDE.

That's all the error messages I received. Thanks again!

**Reply by microdee** [reply]
I have to tell tho, I've spent so much effort into trying to make the plugin ABI compatible with different patch versions but apparently it doesn't work out. Maybe Epic compiles Marketplace plugins every time a new patch version comes out. Oh well, I'll upload a build with 5.3.1 then

**Reply by courcreate** [reply]
Part of the problem is that people always upgrade to the latest version. If it's working for your use case, don't upgrade. Upgrade when there's a new feature that is a must-have for you. This is true for any production environment or manufacturing technique. There's no reason to upgrade my printer firmware if what I'm doing now works great. Same with unreal or any other tool that you are using.

**Reply by onyxlee** [reply]
Thanks, you are right. I shouldn't update if it's not absolutely necessary. But 5.3 has PCG and better lumen and vdb, so I have to.
Your latest compile works for 5.3.1, extremely grateful! Thanks!!!

**Comment by courcreate** [comment]
[](https://github.com/microdee/UE4-SpaceMouse/discussions/99)

Check out this new post and let me know if it works for you @onyxlee
